pulp_file/tests/functional/api/test_auto_publish.py

```diff
@@ -17,94 +16,77 @@
+    gen_object_with_cleanup,
+    file_content_api_client,
+    file_distro_api_client,
+    file_random_content_unit,
+):
+    """Tests auto-publish and auto-distribution"""
+    remote = file_fixture_gen_remote_ssl(manifest_path=basic_manifest_path, policy="on_demand")
+    repo = file_repo_api_client.read(file_repo_with_auto_publish.pulp_href)
+    distribution = gen_object_with_cleanup(
+        file_distro_api_client,
+        {"name": "foo", "base_path": "bar/foo", "repository": repo.pulp_href},
+    )
+
+    # Assert that the repository is at version 0 and that there are no publications associated with
+    # this Repository and that the distribution doesn't have a publication associated with it.
+    assert repo.latest_version_href.endswith("/versions/0/")
+    assert file_pub_api_client.list(repository=repo.pulp_href).count == 0
+    assert file_pub_api_client.list(repository_version=repo.latest_version_href).count == 0
+    assert distribution.publication is None
+
+    # Check what content and artifacts are in the fixture repository
+    expected_files = get_files_in_manifest(remote.url)
+
+    # Sync from the remote
+    body = RepositorySyncURL(remote=remote.pulp_href)
+    monitor_task(file_repo_api_client.sync(repo.pulp_href, body).task)
+    repo = file_repo_api_client.read(repo.pulp_href)
+
+    # Assert that a new repository version was created and a publication was created
+    assert repo.latest_version_href.endswith("/versions/1/")
+    assert file_pub_api_client.list(repository=repo.pulp_href).count == 1
+    assert file_pub_api_client.list(repository_version=repo.latest_version_href).count == 1
+
+    # Assert that the publication has a custom manifest
+    publication = file_pub_api_client.list(repository_version=repo.latest_version_href).results[0]
+    assert publication.manifest == "TEST_MANIFEST"
+
+    # Download the custom manifest
+    files_in_first_publication = get_files_in_manifest(
+        "{}{}".format(distribution.base_url, publication.manifest)
+    )
+    assert files_in_first_publication == expected_files
+
+    # Add a new content unit to the repository and assert that a publication gets created and the
+    # new content unit is in it
+    monitor_task(
+        file_repo_api_client.modify(
+            repo.pulp_href, {"add_content_units": [file_random_content_unit.pulp_href]}
+        ).task
+    )
+    repo = file_repo_api_client.read(repo.pulp_href)
+    files_in_second_publication = get_files_in_manifest(
+        "{}{}".format(distribution.base_url, publication.manifest)
+    )
+    files_added = files_in_second_publication - files_in_first_publication
+    assert repo.latest_version_href.endswith("/versions/2/")
+    assert file_pub_api_client.list(repository=repo.pulp_href).count == 2
+    assert file_pub_api_client.list(repository_version=repo.latest_version_href).count == 1
+    assert len(files_added) == 1
+    assert files_added.pop()[1] == file_random_content_unit.sha256
```

Comment:
Sorry for nitpicking again. `pop` changes the `files_added` and i know that is not a problem programmatically, but semantically i shouldn't. Can you try this (?):
```suggestion
    assert list(files_added)[0][1] == file_random_content_unit.sha256
```

Reply:
I added this change.